Introduce ability to view resolved errors and delete them

acoffman opened the pull request:
Sorry it took so long to circle back around to this! Hopefully with the release of Rails 7.2, backporting a patch will no longer be needed to run this in prod.

This introduces a new collection route at `/errors/resolved` that displays an index view of resolved errors.

On the `show` and `index` views, the `Resolve Error` button will be replaced with a `Delete Error` button when showing resolved errors. The deletion action checks that the error is in fact resolved before proceeding with the deletion.

In addition to the button, I added a small tag next to the error id on the `show` page indicating its status as either `open` or `resolved` so the user has a little more context when on a single error's page.

In order to make this useful, we need some way of navigating to the resolved errors index page (and back). This could be tabs or simply a link or button somewhere. As there is currently no top-level navigation for the site, I wanted to check with you before unilaterally introducing something like that. Maybe just a link for now?

### Screenshots

Error row appearance in resolved index view:
![Screenshot 2024-08-13 at 4 20 10 PM](https://github.com/user-attachments/assets/ff886526-2df6-40b4-bfa8-78084c354a2c)

Show page for an error that has already been resolved:
![Screenshot 2024-08-13 at 4 21 19 PM](https://github.com/user-attachments/assets/8218472c-4a1e-4991-bbef-89a0ff0b1d2e)


closes #4

fractaledmind commented:
This is lovely! Thank you. I'm presently quite busy, but I am trying to carve out time to properly review this and merge and release. But first pass looks good

fractaledmind commented on app/controllers/solid_errors/errors_controller.rb:
```diff
@@ -2,20 +2,16 @@ module SolidErrors
-        .joins(:occurrences)
-        .select(errors_table[Arel.star],
-          occurrences_table[:created_at].maximum.as("recent_occurrence"),
-          occurrences_table[:id].count.as("occurrences_count"))
-        .group(errors_table[:id])
-        .order(recent_occurrence: :desc)
+    # GET /errors/resolved
+    def resolved
+      @errors = get_errors(base_scope: Error.resolved)
```
I would prefer using a query param for the scope, which has a default. Something like:
```ruby
def index
  errors_table = Error.arel_table
  occurrences_table = Occurrence.arel_table
  scope = params[:scope] == "resolved" ? Error.resolved : Error.unresolved
  @errors = scope
    .joins(:occurrences)
    .select(errors_table[Arel.star],
      occurrences_table[:created_at].maximum.as("recent_occurrence"),
      occurrences_table[:id].count.as("occurrences_count"))
    .group(errors_table[:id])
    .order(recent_occurrence: :desc)
end

fractaledmind replied:
I don't like non-RESTful actions, plus it will make the path helpers easier

fractaledmind commented on config/routes.rb:
```diff
@@ -1,5 +1,9 @@
   get "/" => "errors#index", :as => :root
 
-  resources :errors, only: [:index, :show, :update], path: ""
+  resources :errors, only: [:index, :show, :update, :destroy], path: "" do
+    collection do
+      get :resolved
+    end
+  end
```
When we collapse both views into the index route, we won't need this custom route.

fractaledmind reviewed (commented):
Some thoughts.

acoffman commented:
Thanks for the review! 

I have pushed a commit that I believe addresses all of your feedback. Additionally I have pushed a second commit that adds a toggle to the top right of the index page allowing you to switch between resolved and unresolved index views. Screenshot below.

![Screenshot 2024-08-16 at 11 56 20 AM](https://github.com/user-attachments/assets/81bfe92c-37e0-4b9d-8eae-7176c5166d87)

fractaledmind commented:
Lovely. I want to make a few small tweaks. I'll try to carve out time next week. This is excellent tho. Thank you for this!

fractaledmind commented:
Released with version 0.5.0
